app/controllers/api/storage.php

```diff
@@ -138,6 +135,110 @@
+        /** @var Appwrite\Utopia\Response $response */
+        /** @var Utopia\Database\Database $dbForInternal */
+        /** @var Appwrite\Event\Event $audits */
+
+        $bucket = $dbForInternal->getDocument('buckets', $bucketId);
+
+        if (empty($bucket->getId())) {
+            throw new Exception('Bucket not found', 404);
+        }
+
+        $read = (is_null($read)) ? $bucket->getAttribute('$read', []) : $read; // By default inherit read permissions
+        $write = (is_null($write)) ? $bucket->getAttribute('$write',[]) : $write; // By default inherit write permissions
+
+        $bucket = $dbForInternal->updateDocument('buckets', $bucket->getId(), $bucket
+                ->setAttribute('name',$name)
+                ->setAttribute('$read',$read)
+                ->setAttribute('$write',$write)
+                ->setAttribute('maximumFileSize',$maximumFileSize)
+                ->setAttribute('allowedFileExtensions',$allowedFileExtensions)
+                ->setAttribute('enabled',$enabled)
+                ->setAttribute('adapter',$adapter)
+                ->setAttribute('encryption',$encryption)
+                ->setAttribute('antiVirus',$antiVirus)
+        );
+
+        $audits
+            ->setParam('event', 'storage.buckets.update')
+            ->setParam('resource', 'storage/buckets/' . $bucket->getId())
+            ->setParam('data', $bucket->getArrayCopy())
+        ;
+
+        $response->dynamic2($bucket, Response::MODEL_BUCKET);
+    });
+
+App::delete('/v1/storage/buckets/:bucketId')
+    ->desc('Delete Bucket')
+    ->groups(['api', 'storage'])
+    ->label('scope', 'buckets.write')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'storage')
+    ->label('sdk.method', 'deleteBucket')
+    ->label('sdk.description', '/docs/references/storage/delete-bucket.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_NOCONTENT)
+    ->label('sdk.response.model', Response::MODEL_NONE)
+    ->param('bucketId', '', new UID(), 'Bucket unique ID.')
+    ->inject('response')
+    ->inject('dbForInternal')
+    ->inject('audits')
+    ->inject('deletes')
+    ->action(function ($bucketId, $response, $dbForInternal, $audits, $deletes) {
+        /** @var Appwrite\Utopia\Response $response */
+        /** @var Utopia\Database\Database $dbForInternal */
+        /** @var Appwrite\Event\Event $audits */
+        /** @var Appwrite\Event\Event $deletes */
+
+        $bucket = $dbForInternal->getDocument('buckets', $bucketId);
+
+        if (empty($bucket->getId())) {
+            throw new Exception('Bucket not found', 404);
+        }
+
+        $deletes
+            ->setParam('type', DELETE_TYPE_DOCUMENT)
+            ->setParam('document', $bucket)
+        ;
+
+        if(!$dbForInternal->deleteDocument('buckets', $bucketId)) {
+            throw new Exception('Failed to remove project from DB', 500);
+        }
+
+        $audits
```

Comment:
We should see what is the most performant way to delete a bucket for each adapter. We obviously need to delete all the files from the internalDB, but how do we delete the files? a single call to s3? the full parent directory?

Reply:
Whatever we decide, the delete for bucket should be handled by the delete worker right?

Reply:
Yes. Let's discuss again when we get there.